db/rdbms/src/main/java/io/camunda/db/rdbms/write/queue/ExecutionQueue.java

```diff
@@ -53,56 +55,100 @@ public void registerPostFlushListener(final PostFlushListener listener) {
     postFlushListeners.add(listener);
   }
 
-  public void flush() {
-    if (queue.isEmpty()) {
-      LOG.trace(
-          "[RDBMS ExecutionQueue, Partition {}] Skip Flushing because execution queue is empty",
-          partitionId);
-      return;
-    }
-    LOG.debug(
-        "[RDBMS ExecutionQueue, Partition {}] Flushing execution queue with {} items",
-        partitionId,
-        queue.size());
-
-    final var startMillis = System.currentTimeMillis();
-    final var session =
-        sessionFactory.openSession(ExecutorType.BATCH, TransactionIsolationLevel.READ_UNCOMMITTED);
-
-    var flushedElements = 0;
-    try {
-      while (!queue.isEmpty()) {
-        final var entry = queue.peek();
-        LOG.trace("[RDBMS ExecutionQueue, Partition {}] Executing entry: {}", partitionId, entry);
-        session.update(entry.statementId(), entry.parameter());
-        queue.remove();
-        flushedElements++;
+  /**
+   * Performs flush on the queue.
+   *
+   * @return number of flushed items
+   */
+  public int flush() {
+    synchronized (queue) {
+      if (queue.isEmpty()) {
+        LOG.trace(
+            "[RDBMS ExecutionQueue, Partition {}] Skip Flushing because execution queue is empty",
+            partitionId);
+        return 0;
       }
-
-      if (!preFlushListeners.isEmpty()) {
-        LOG.debug("[RDBMS ExecutionQueue, Partition {}] Call pre flush listeners", partitionId);
-        preFlushListeners.forEach(PreFlushListener::onPreFlush);
+      LOG.debug(
+          "[RDBMS ExecutionQueue, Partition {}] Flushing execution queue with {} items",
+          partitionId,
+          queue.size());
+
+      final var startMillis = System.currentTimeMillis();
+      final var session =
+          sessionFactory.openSession(
+              ExecutorType.BATCH, TransactionIsolationLevel.READ_UNCOMMITTED);
+
+      var flushedElements = 0;
+      try {
+        while (!queue.isEmpty()) {
+          final var entry = queue.peek();
+          LOG.trace("[RDBMS ExecutionQueue, Partition {}] Executing entry: {}", partitionId, entry);
+          session.update(entry.statementId(), entry.parameter());
```

Comment:
Hm I know, we also had it before, but is it correct to always do an update? Although we also have inserts? (Don't know what difference it makes for mybatis ... )

Reply:
Funny thing: MyBatis forwards all insert() calls to update() 

`  @Override
  public int insert(String statement, Object parameter) {
    return update(statement, parameter);
  }
`